internal/extsvc/gitlab/merge_requests.go

```diff
@@ -135,6 +136,9 @@ func (c *Client) CreateMergeRequest(ctx context.Context, project *Project, opts
 		if code == http.StatusConflict {
 			return nil, ErrMergeRequestAlreadyExists
 		}
+		if code <= 400 && code < 200 {
```

Comment:
```suggestion
		if code <= http.StatusBadRequest && code < http.StatusOK {
```

How about making use of the exported code from the `http` package?

Reply:
Depending on what we decide re:[my more general comment about which statuses we do/don't retry](https://github.com/sourcegraph/sourcegraph/pull/49919#issuecomment-1485803374), I might suggest we define a very small helper somewhere that we could leverage in all of these places where we do the status code check, something like:

```go
func maybeMakeNonRetryable(statusCode number, err error) error {
	if [code matches statuses we shouldn't return] {
		return errcode.MakeNonRetryable(err)
	}
	return err
}
```

So it takes the `statusCode` and the original `err`, looks at the status code, and either
- returns the original `err` if the status code is one we'd typically retry
- returns `errcode.MakeNonRetryable(err)` if it's one we'd typically _not_ retry

Reply:
Thanks for addressing some of this feedback! What did you think about this last suggestion 👆, to pull the status code checking logic out into a helper function that could be leveraged in each of the code host files?

Reply:
Yes, that'd make this cleaner and reusable in the future. I agree let's create that helper.

Reply:
Yeah i think its a great idea, thanks! thinking now about where i would put it. doesnt seem like it'd just be one place due to the way we handle the different code hosts @courier-new

Reply:
The logic for figuring out non-retryable errors is the same for all code hosts, yes?

If yes, then it could live in `internal/extsvc/common` or something in the root of `internal/extsvc` so that all the external services can import it.

Reply:
That's a great question! Since ultimately this helper is specific to worker automatic retries, my first thought was it might make sense to put in the worker package code somewhere. But, given we know well that Go doesn't like import cycles, that might not be the best place to import from across the external service code. In light of that, the next best place that might make sense would probably be the same place where `MakeNonRetryable` is defined in the `errcode` package.

Reply:
Oops missed that Bolaji replied at the same time. That's also a good suggestion. I think you can't go wrong either way. 🙂